api/envoy/extensions/filters/http/oauth/v3/oauth.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.oauth.v3;
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.oauth.v3";
+option java_outer_classname = "OauthProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: OAuth]
+// OAuth :ref:`configuration overview <config_http_filters_oauth>`.
+// [#extension: envoy.filters.http.oauth]
+//
+
+message OAuth2Credentials {
+  // The client_id and client_secret should be provided by the OAuth provider.
+  string client_id = 1;
+
+  string client_secret = 2;
+
+  string token_secret = 3;
+}
+
+// OAuth filter config
+//
+// [#next-free-field: 11]
+message OAuth2 {
+  // The cluster that has the SDS generic secrets.
+  string secrets_cluster = 1 [(validate.rules).string = {min_bytes: 1}];
```

Comment:
What a generic secret type will be provided from this cluster? OAuth2Credentials?